scientific/edf.ksy

```diff
@@ -0,0 +1,182 @@
+        size: 80
+        type: str
+        encoding: ASCII
+
+      - id: "start_date"
+        size: 8
+        type: str
+        encoding: ASCII
+
+      - id: "start_time"
+        size: 8
+        type: str
+        encoding: ASCII
+
+      - id: "header_num_bytes"
+        size: 8
+        type: str
+        encoding: ASCII
+        pad-right: 0x20
+
+      - id: "reserved"
+        size: 44
+        type: str
+        encoding: ASCII
+
+      - id: "num_data_records"
+        size: 8
+        type: str
+        pad-right: 0x20
+        encoding: ASCII
+
+      - id: "data_duration"
+        size: 8
+        type: str
+        pad-right: 0x20
+        encoding: ASCII
+
+      - id: "num_signals"
+        size: 4
+        type: str
+        pad-right: 0x20
+        encoding: ASCII
+
+      - id: "labels"
+        size: 16
+        repeat: expr
+        repeat-expr: ns
+        type: str
+        pad-right: 0x20
+        encoding: ASCII
+
+      - id: "transducers"
+        size: 80
+        repeat: expr
+        repeat-expr: ns
+        type: str
+        pad-right: 0x20
+        encoding: ASCII
+
+      - id: "phys_dimensions"
+        size: 8
+        repeat: expr
+        repeat-expr: ns
+        type: str
+        pad-right: 0x20
+        encoding: ASCII
+
+      - id: "phys_mins"
+        size: 8
+        repeat: expr
+        repeat-expr: ns
+        type: str
+        pad-right: 0x20
+        encoding: ASCII
+
+      - id: "phys_maxes"
+        size: 8
+        repeat: expr
+        repeat-expr: ns
+        type: str
+        pad-right: 0x20
+        encoding: ASCII
+
+      - id: "dig_mins"
+        size: 8
+        repeat: expr
+        repeat-expr: ns
+        type: str
+        pad-right: 0x20
+        encoding: ASCII
+
+      - id: "dig_maxes"
+        size: 8
+        repeat: expr
+        repeat-expr: ns
+        type: str
+        pad-right: 0x20
+        encoding: ASCII
+
+      - id: "prefiltering"
+        size: 80
+        repeat: expr
+        repeat-expr: ns
+        type: str
+        pad-right: 0x20
+        encoding: ASCII
+
+      - id: "num_samples"
+        size: 8
+        repeat: expr
+        repeat-expr: ns
+        type: str
+        pad-right: 0x20
+        encoding: ASCII
+
+      - id: "signals_reserved"
+        size: 32
+        repeat: expr
+        repeat-expr: ns
+        type: str
+        pad-right: 0x20
+        encoding: ASCII
+
+  body:
+    types:
+      data_recording:
+        seq:
+          - id: "data"
+            type: u2
+            repeat: expr
+            repeat-expr: _root.header.num_samples[_parent.signal_index].to_i
+
+      signal:
+        params:
+          - id: "signal_index"
+            type: s4
+
+        seq:
+          - id: "recordings"
+            repeat: expr
+            repeat-expr: _root.header.ndr
+            type: data_recording
+
+    seq:
+      - id: "signals"
+        repeat: expr
+        repeat-expr: _root.header.ns
+        type: signal(_index)
```

Comment:
`seq` must come before `types` in the type specification - see https://doc.kaitai.io/ksy_style_guide.html#type:

> ## 2. Order of sections in a type spec
> 
> Use the following order of sections:
> 
> - `meta`, if present, MUST go first
> - `doc`
> - `doc-ref`
> - <ins>**`seq`**</ins>
> - `instances`, <ins>**`types`**</ins>, `enums` — use one’s best judgement to order these 3 to maximize readability

This is what searching for a name of the type in which a `seq` structure is located looks like in practice if this rule is not followed ([`font/ttf.ksy:338-650`](https://github.com/kaitai-io/kaitai_struct_formats/blob/ff19ec06d5fcba3ea382ed7acdcc462b866ed154/font/ttf.ksy#L338-L650)):

![ksf-ttf-type-spec-bad-order](https://user-images.githubusercontent.com/47499687/153732817-ffc77d7f-a3a1-4385-bc1f-f6889578ffca.gif)